alf/experience_replayers/replay_buffer.py

```diff
@@ -286,49 +331,65 @@ def get_batch(self, batch_size, batch_length):
             result = alf.nest.map_structure(lambda b: b[(out_env_ids, idx)],
                                             self._buffer)
 
-            if self._prioritized_sampling:
-                indices = self._env_id_idx_to_index(env_ids,
-                                                    self.circular(start_pos))
-                avg_weight = self._sum_tree.nnz / self._sum_tree.summary()
-                info = info._replace(
-                    importance_weights=self._sum_tree[indices] * avg_weight)
-
             if self._postprocess_exp_fn:
                 result, info = self._postprocess_exp_fn(self, result, info)
 
         return convert_device(result), convert_device(info)
 
+    def _recent_sample(self, batch_size, batch_length):
+        return self._sample(batch_size, batch_length, self._recent_data_steps)
+
     def _uniform_sample(self, batch_size, batch_length):
         min_size = self._current_size.min()
         assert min_size >= batch_length, (
             "Not all environments have enough data. The smallest data "
             "size is: %s Try storing more data before calling get_batch" %
             min_size)
+        return self._sample(batch_size, batch_length)
 
+    def _sample(self, batch_size, batch_length, recent_data_steps=None):
```

Comment:
rename this arg to something like sample_from_recent_n_steps, so it's more readable.

Reply:
changed